lib/openhab/dsl/items/builder.rb

```diff
@@ -288,6 +291,7 @@ def initialize(type, name = nil, label = nil,
                        state: nil)
           raise ArgumentError, "`name` cannot be nil" if name.nil?
           raise ArgumentError, "`dimension` can only be specified with NumberItem" if dimension && type != :number
+          raise ArgumentError, "`thing` was given without `channel`" if @thing && channel.nil?
```

Comment:
I don't think this should be an argument error. they may provide (multiple) channel(s) in the block, and the thing should apply to all of them

Reply:
Good point! I didn't think of that scenario! I've removed this check and also added a test for mixed case.